react/features/authentication/functions.any.ts

```diff
@@ -0,0 +1,45 @@
+import { IConfig } from '../base/config/configType';
+import { getBackendSafeRoomName } from '../base/util/uri';
+
+/**
+ * Checks if the token for authentication is available.
+ *
+ * @param {Object} config - Configuration state object from store.
+ * @returns {boolean}
+ */
+export const isTokenAuthEnabled = (config: IConfig): boolean =>
+    typeof config.tokenAuthUrl === 'string' && config.tokenAuthUrl.length > 0;
+
+/**
+ * Returns the state that we can add as a parameter to the tokenAuthUrl.
+ *
+ * @param {string?} roomName - The room name.
+ * @param {string?} tenant - The tenant name if any.
+ * @param {boolean} skipPrejoin - Whether to skip pre-join page.
+ * @returns {Object} The state object.
+ */
+export const _getTokenAuthState = (
+        roomName: string | undefined,
+        tenant: string | undefined,
+        skipPrejoin: boolean | undefined = false): object => {
+    const state = {
+        room: roomName,
+        roomSafe: getBackendSafeRoomName(roomName),
+        tenant
+    };
+
+    if (skipPrejoin) {
+        // We have already shown the prejoin screen, no need to show it again after obtaining the token.
+        // @ts-ignore
+        state['config.prejoinConfig.enabled'] = false;
+    }
+
+    const params = new URLSearchParams(window.location.search);
```

Comment:
I think we should parse and group the parameters in `config` and `interfaceConfig` back, so we can serialize them easily in the Firebase app, instead of blindly taking whatever the user puts in the URL.

Reply:
I don't get this? What is your proposal exactly?

Reply:
Right now if the URL has `?foo=bar` we are going to pass it around, which is weird no? So I was suggesting that if the URL has `config.startWithAudioOonly=true` for example, we'd turn that into:

```
config: {
    startWithAudioOnly: true
}
```

And then on the other side to the opposite.

The funny thing is, as I type this... I think it's nonsense.

So maybe we should just filter was we pass, and only take things that start with `config.` or `interfaceConfig.`

Reply:
Ok, I will add that filtering